### PR TITLE
refactor(dataset): finish dataset dispatch refactor

### PR DESCRIPTION
**Pull request by ramfox**
Finish dataset dispatch refactor
* [x] Rename
* [x] Remove
* [x] Pull
* [x] Validate
* [x] Manifest
* [x] ManifestMisssing
* [x] DAGInfo
* [x] Stats 

depends on #1701

### CODE REVIEW

File: lib/api.go [hunk 1]
```diff
@@ -65,6 +65,9 @@ const (
 
 	// dataset endpoints
 
+	// AEListRawl lists all datasets in your collection as a well formatted string
+	AEListRaw = APIEndpoint("/listraw")
+
```
Comment (Arqu) [inline lib/api.go:70]: `AEListRawl > AEListRaw` in the comment
Also don't need the empty line.

File: api/util/errors.go [hunk 2]
```diff
@@ -71,3 +77,9 @@ func RespondWithError(w http.ResponseWriter, err error) {
 	WriteErrResponse(w, http.StatusInternalServerError, err)
 	return
 }
+
+// RespondWithDispatchTypeError(w http.ResponseWriter, got
+func RespondWithDispatchTypeError(w http.ResponseWriter, got interface{}) {
+	log.Errorf("type mismatch: %v of type %s", got, reflect.TypeOf(got))
+	WriteErrResponse(w, http.StatusInternalServerError, fmt.Errorf("type mismatch: %v of type %s", got, reflect.TypeOf(got)))
```
Comment (Arqu) [inline api/util/errors.go:84]: 👍

File: cmd/access.go [hunk 3]
```diff
@@ -60,8 +60,8 @@ type AccessOptions struct {
 
 // Complete adds any missing configuration that can only be added just before calling Run
 func (o *AccessOptions) Complete(f Factory, args []string) (err error) {
-	o.Instance = f.Instance()
-	return nil
+	o.Instance, err = f.Instance()
+	return
```
Comment (Arqu) [inline cmd/access.go:64]: can drop the return here

File: cmd/checkout.go [hunk 4]
```diff
@@ -49,24 +49,21 @@ type CheckoutOptions struct {
...
 	}
 
 	if len(args) == 2 {
 		o.Dir = args[1]
 	} else {
 		o.Dir = ""
 	}
-
-	return nil
+	return
```
Comment (Arqu) [inline cmd/checkout.go:66]: Can drop the return, but in general I'm not a huge fan of named return values

**Review by Arqu — APPROVED** [review]
LGTM. Some nits, but great stuff. `lib/datasets.go` is quite a mouthful.

File: cmd/access.go (hunk 3 above)
Comment (b5) [reply cmd/access.go:64]: always use an explicit return value. It's fine to use named return values as documentation, but not as default returns

**Pull request merged**